roles/test-utils/mining-device/src/main.rs

```diff
@@ -669,3 +674,70 @@ fn generate_random_32_byte_array() -> [u8; 32] {
     rng.fill(&mut arr[..]);
     arr
 }
+
+fn start_mining_threads(
+    have_new_job: Receiver<()>,
+    miner: Arc<Mutex<Miner>>,
+    share_send: Sender<(u32, u32, u32, u32)>,
+) {
+    tokio::task::spawn(async move {
+        let mut killers: Vec<Arc<AtomicBool>> = vec![];
+        loop {
```

Comment:
nit: maybe we could use only `tokio` in this function with https://docs.rs/tokio/latest/tokio/runtime/#multi-thread-scheduler

Reply:
my plan was to dedicate only one thread for the async runtime (I forgot to do it) and have one dedicated thread for each miner. I think is better like that cause miners thread have no blocking operation. I would stick with my plan. If you have any particular reason in mind to no do it pls let me know.

Reply:
I'm going to dedicate only one thread to tokio runtime and all the others to mining. I don't see any advantage in using work stealing since mining thread are constantly doing something. Mark as resolved for now, let me know if you are ok with that